platform/android/MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/attribution/AttributionParseTest.java

```diff
@@ -0,0 +1,309 @@
+package com.mapbox.mapboxsdk.attribution;
+
+import com.mapbox.mapboxsdk.BuildConfig;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.Set;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class)
+public class AttributionParseTest {
+
+  private static final String STREETS_ATTRIBUTION = "<a href=\"https://www.mapbox.com/about/maps/\" target=\"_blank\">&copy; Mapbox</a> <a href=\"http://www.openstreetmap.org/about/\" target=\"_blank\">&copy; OpenStreetMap</a> <a class=\"mapbox-improve-map\" href=\"https://www.mapbox.com/map-feedback/\" target=\"_blank\">Improve this map</a>\n";
+  private static final String SATELLITE_ATTRIBUTION = "<a href=\"https://www.mapbox.com/about/maps/\" target=\"_blank\">&copy; Mapbox</a> <a href=\"http://www.openstreetmap.org/about/\" target=\"_blank\">&copy; OpenStreetMap</a> <a class=\"mapbox-improve-map\" href=\"https://www.mapbox.com/map-feedback/\" target=\"_blank\">Improve this map</a> <a href=\"https://www.digitalglobe.com/\" target=\"_blank\">&copy; DigitalGlobe</a>\n";
+
+  @Test
+  public void testParseAttributionStringSatellite() throws Exception {
+    AttributionParser attributionParser = new AttributionParser.Options()
+      .withAttributionData(SATELLITE_ATTRIBUTION)
+      .build();
+
+    Set<Attribution> attributionList = attributionParser.getAttributions();
+    assertEquals("Size of list should match", 4, attributionList.size());
+
+    int counter = 0;
+    for (Attribution attribution : attributionList) {
+      switch (counter) {
+        case 0:
+          assertEquals("URL mapbox should match", "https://www.mapbox.com/about/maps/", attribution.getUrl());
```

Comment:
What about extracting the different assertions into different tests and give them a name based on the messages?
This way I guess that the `for` loop won't be necessary and tests will be more maintainable and readable.

Reply:
don't have bandwith to pick this up. feel free to ticket out any follow up work.